src/Generator/ClassFinder.php

```diff
@@ -0,0 +1,213 @@
+<?php
+declare(strict_types=1);
+
+namespace LoyaltyCorp\ApiDocumenter\Generator;
+
+use DateTime as BaseDatetime;
+use DateTimeInterface;
+use EoneoPay\Utils\DateTime;
+use EoneoPay\Utils\UtcDateTime;
+use Symfony\Component\PropertyInfo\Extractor\PhpDocExtractor;
+use Symfony\Component\PropertyInfo\Extractor\ReflectionExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractor;
+use Symfony\Component\PropertyInfo\PropertyInfoExtractorInterface;
+use Symfony\Component\PropertyInfo\Type;
+
+final class ClassFinder
+{
+    /**
+     * @var PropertyInfoExtractorInterface
+     */
+    private $propertyInfo;
+
+    /**
+     * @var string[]
+     */
+    private $skipClasses;
+
+    /**
+     * Constructor.
+     *
+     * @param string[] $skipClasses
+     */
+    public function __construct(
+        array $skipClasses
+    ) {
+        // This is constructed and not injected because it is not intended to be customisable.
+        // The extractors are configured explicitly for the purposes of finding classes.
+        $phpDocExtractor = new PhpDocExtractor();
+        $reflectionExtractor = new ReflectionExtractor(
+            null,
+            null,
+            null,
+            true,
+            ReflectionExtractor::ALLOW_PRIVATE |
+                ReflectionExtractor::ALLOW_PROTECTED |
+                ReflectionExtractor::ALLOW_PUBLIC
+        );
+
+        $this->propertyInfo = new PropertyInfoExtractor(
```

Comment:
This sucks, but I didnt want a container definition to have the power to change the behaviour of this class. I guess a static closure in the service provider could still achieve the same outcome.

Reply:
> This sucks, but I didnt want a container definition to have the power to change the behaviour of this class.
Why not?

Reply:
The primary feeling here is that the ClassFinder is designed for a single purpose and its only "specified" for behaviour with a specific configuration of PropertyInfo.

I'll inject it and just make sure the service provider configures it correctly.